### PR TITLE
Adj matrix

### PR DESCRIPTION
**Pull request by johann-cm**
Add features to the AdjacencyMatrix trait

### CODE REVIEW

File: src/datastructures/nalgebra.rs [hunk 1]
```diff
@@ -20,16 +36,40 @@ impl From<Graph> for NAMatrix {
...
             }
         }
-        res
+        NAMatrix(res)
     }
 }
 
 impl AdjacencyMatrix for NAMatrix {
+    fn from_dim(dim: usize) -> Self {
+        NAMatrix(DMatrix::from_element(dim, dim, f64::INFINITY))
```
Comment (lquenti) [inline src/datastructures/nalgebra.rs]: are you sure that this is true?
It seems that if I create with `from_dim` and then set each vertices through `set`, the diagonal would be `f64::INFINITY`, which is wrong as it should be 0.

File: src/parser.rs [hunk 2]
```diff
@@ -142,6 +156,45 @@ impl From<&Graph> for WeightedGraph {
     }
 }
 
+impl AdjacencyMatrix for Graph {
```
Comment (lquenti) [inline src/parser.rs]: Do we `really` want to offer this?

I feel like most most algorithms are optimized for either representation, matrix or list. But pretending that a Graph is a Adjacency**Matrix** seems like an invitation to write bad code.

Why do you want this? Especially my exact algorithms where I am doing that many cycle traversals are really really not made for adjacancy lists.

**Review by lquenti — CHANGES_REQUESTED** [review]
other than that nice code

File: src/datastructures/nalgebra.rs (hunk 1 above)
Comment (johann-cm) [reply src/datastructures/nalgebra.rs]: You are right, I think resolved it now

File: src/parser.rs (hunk 2 above)
Comment (johann-cm) [reply src/parser.rs]: Ok, that is a valid point. I removed it.

**Review by lquenti — APPROVED** [review]

**Pull request merged**